### PR TITLE
Update frequently-asked-questions.md

### PR DESCRIPTION
**Pull request by josevalim**
Explanations inline but feel free to discard them.

### CODE REVIEW

File: frequently-asked-questions.md [hunk 1]
```diff
@@ -177,24 +177,26 @@ It is very popular and a great language!
...
 - Elixir has a namespace for module functions and another for variables,
   Gleam has one unified namespace (so there's no special `fun.()` syntax).
 - Gleam standard library is distributed as Hex packages, which makes interoperability
   with other BEAM languages easier.
 - Elixir is a larger language, featuring numerous language features not present
   in Gleam.
-- Elixir has an excellent official test framework, Gleam does not yet.
+- Elixir has a robust test framework with excellent support for concurrency,
+  partitioning, parameterized tests, integrated error reports, and more.
+  Gleam does not yet.
```
Comment (josevalim) [inline frequently-asked-questions.md:193]: I don't know if Gleam does concurrency in its unit testing framework

File: frequently-asked-questions.md [hunk 2]
```diff
@@ -177,24 +177,26 @@ It is very popular and a great language!
...
 - Both languages compile to Erlang but Elixir compiles to Erlang abstract
   format, while Gleam compiles to Erlang source. Gleam can also compile to
   JavaScript.
-- Elixir has superior BEAM runtime integration, featuring perfectly accurate
+- Elixir has superior BEAM runtime integration, featuring accurate
```
Comment (josevalim) [inline frequently-asked-questions.md:197]: Unfortunately tail call optimizations make it impossible for any of Gleam or Elixir to be perfect.

File: frequently-asked-questions.md [hunk 3]
```diff
@@ -177,24 +177,26 @@ It is very popular and a great language!
...
 Here's a non-exhaustive list of differences:
 
-- Elixir is dynamically typed, Gleam is statically typed. Elixir is developing a
-  novel, but it has not yet released.
+- Elixir is dynamically typed, Gleam is statically typed. Elixir is integrating a
+  gradual type system into the language, but it has no user facing features yet.
 - Elixir has a powerful macro system, Gleam has no metaprogramming features.
 - Elixir's compiler is written in Erlang and Elixir, Gleam's is written in Rust.
-- Elixir uses Ruby style syntax, Gleam has a C family style syntax.
+- Gleam has a more traditional C family style syntax.
```
Comment (josevalim) [inline frequently-asked-questions.md:184]: I am not sure how much I'd describe Elixir as Ruby style. A possible suggestion is to emphasize Gleam being more traditional / familiar / etc. Your call.

File: frequently-asked-questions.md (hunk 1 above)
Comment (lpil) [reply frequently-asked-questions.md:193]: We don't have a canonical framework at all. The most commonly use one is eunit run without any concurrency

File: frequently-asked-questions.md (hunk 2 above)
Comment (lpil) [reply frequently-asked-questions.md:197]: Thanks

File: frequently-asked-questions.md (hunk 3 above)
Comment (lpil) [reply frequently-asked-questions.md:184]: Seems reasonable to me

**Review by lpil — APPROVED** [review]

**Comment by lpil** [comment]
Thank you!!

**Pull request merged**